### ISSUE TITLE
[NativeAOT] Generic virtual method pointer lookup failure

### ISSUE THREAD

**Issue by dellamonica** [dotnet/runtimelab#1473]
@MichalStrehovsky Follow-up for issue #1455

Repro:

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net6.0</TargetFramework>
    <Nullable>enable</Nullable>
    <!-- Removing these next two causes the bug to go away -->
    <IlcDisableReflection>true</IlcDisableReflection>
    <IlcGenerateStackTraceData>false</IlcGenerateStackTraceData>
  </PropertyGroup>

  <ItemGroup>
    <PackageReference Include="Microsoft.DotNet.ILCompiler" Version="7.0.0-*" />
  </ItemGroup>

</Project>
```

Code

```csharp
var list = new List<string>() { "ABC", "X", "XYZ", "UV", "XA" };
var ordered = list.OrderBy(s => s.Length).ThenBy(s => s.GetHashCode());
foreach (var s in ordered)
{
    Console.WriteLine(s);
}
```

**Comment by MichalStrehovsky** [comment]
Thanks for the good repro! The fix is in #1477.

**Comment by dellamonica** [comment]
Thank you for the quick fix. Once the NuGet package is published I'll give it a try and test it.

**Comment by MichalStrehovsky** [comment]
Fixed in #1477
